discovery/manager_test.go

```diff
@@ -774,6 +774,49 @@ scrape_configs:
 	verifyPresence(discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"bar:9090\"}", false)
 }
 
+func TestApplyConfigDoesNotModifyStaticProviderTargets(t *testing.T) {
+	cfgText := `
+scrape_configs:
+ - job_name: 'prometheus'
+   static_configs:
+   - targets: ["foo:9090"]
+   - targets: ["bar:9090"]
+`
+	originalConfig := &config.Config{}
+	if err := yaml.UnmarshalStrict([]byte(cfgText), originalConfig); err != nil {
+		t.Fatalf("Unable to load YAML config cfgYaml: %s", err)
+	}
+	origScrpCfg := originalConfig.ScrapeConfigs[0]
+
+	processedConfig := &config.Config{}
+	if err := yaml.UnmarshalStrict([]byte(cfgText), processedConfig); err != nil {
+		t.Fatalf("Unable to load YAML config cfgYaml: %s", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	discoveryManager := NewManager(ctx, nil)
+	go discoveryManager.Run()
+
+	c := make(map[string]sd_config.ServiceDiscoveryConfig)
+	for _, v := range processedConfig.ScrapeConfigs {
+		c[v.JobName] = v.ServiceDiscoveryConfig
+	}
+	discoveryManager.ApplyConfig(c)
+
+	select {
+	case tgsmap, ok := <-discoveryManager.SyncCh():
+		if !ok {
+			t.Fatalf("problem getting target grouops from channel\n")
+		}
+		for _, tgs := range tgsmap {
+			if !reflect.DeepEqual(origScrpCfg.ServiceDiscoveryConfig.StaticConfigs, tgs) {
+				t.Fatalf("discovery manager modified static config \n  expected: %v\n  got: %v\n",
+					origScrpCfg.ServiceDiscoveryConfig.StaticConfigs, tgs)
+			}
+		}
+	}
+}
```

Comment:
The test fails in Travis because there's no guarantee on the order of targets. What about just waiting for the discovery manager to send the updates and check that it didn't touch the configuration?

```go
<-discoveryManager.SyncCh()
if !reflect.DeepEqual(origScrpCfg.ServiceDiscoveryConfig.StaticConfigs, processedConfig...) {
...
}
```